### PR TITLE
Adding Jazzy doc for SalesforceSwiftSDK and other doc related cleanup

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: build/build_doc.xml [hunk 1]
```diff
@@ -66,16 +66,6 @@
             <param name="PROJECT_NAME" value="SmartSync" />
         </antcall>
     </target>
-    
-    <target name="swiftDoc">
```
Comment (bhariharan) [inline build/build_doc.xml]: This target wasn't doing anything.

File: docs/generate_docs.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+#
+# Run this script from the root of the repo to generate docs for the libraries in this workspace.
```
Comment (bhariharan) [inline docs/generate_docs.sh]: A simple script that does both `Appledoc` for `Obj-C` and `Jazzy` for `Swift`.

File: docs/jazzy.yaml [hunk 3]
```diff
@@ -8,15 +8,11 @@ exclude:
...
+swift_version: 4.0.3
 skip_undocumented: true
-
 theme: fullwidth
-
 clean: false
-
-xcodebuild_arguments: [-configuration, Release,-sdk, iphonesimulator,-scheme, SalesforceSwiftSDK, -workspace, SalesforceMobileSDK.xcworkspace]
+xcodebuild_arguments: [-configuration, Release, -sdk, iphonesimulator, -scheme, SalesforceSwiftSDK, -workspace, SalesforceMobileSDK.xcworkspace]
+output: ../build/artifacts/doc/SalesforceSwiftSDK
```
Comment (bhariharan) [inline docs/jazzy.yaml]: Adding output directory argument to be consistent with where the other library docs reside.

File: libs/SalesforceSwiftSDK/readme.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 Mobile SDK enhances its Swift offering with PromiseKit adaptations. You can now use promises for these SDK components:
 
-* [SalesforceSwiftSDKManager](Classes/SalesforceSwiftSDKManager.html)
-* [SFRestAPI](Extensions/SFRestAPI.html)
-* [SFSmartStoreClient](Classes/SFSmartStoreClient.html)
-* [SFSmartSyncSyncManager](Extensions/SFSmartSyncSyncManager.html)
-* [SFUserAccountManager](Extensions/SFUserAccountManager.html)
+* [SalesforceSwiftSDKManager](http://forcedotcom.github.io/SalesforceMobileSDK-iOS/Documentation/SalesforceSwiftSDK/Classes/SalesforceSwiftSDKManager.html)
```
Comment (bhariharan) [inline libs/SalesforceSwiftSDK/readme.md]: Fixing links.

File: readme.md [hunk 5]
```diff
@@ -84,6 +84,7 @@ Documentation
 * [SmartSync Library Reference](http://forcedotcom.github.io/SalesforceMobileSDK-iOS/Documentation/SmartSync/html/index.html)
 * [SalesforceHybridSDK Library Reference](http://forcedotcom.github.io/SalesforceMobileSDK-iOS/Documentation/SalesforceHybridSDK/html/index.html)
 * [SalesforceReact Library Reference](http://forcedotcom.github.io/SalesforceMobileSDK-iOS/Documentation/SalesforceReact/html/index.html)
+* [SalesforceSwiftSDK Library Reference](http://forcedotcom.github.io/SalesforceMobileSDK-iOS/Documentation/SalesforceSwiftSDK/index.html)
```
Comment (bhariharan) [inline readme.md]: Adding link to new library's doc.

**Comment by rwhitley** [comment]
👍

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please re-submit this PR to dev, we may have already fixed your issue.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_PR-Check">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**